packages/polkadart_scale_codec/test/source_test.dart

```
@@ -0,0 +1,252 @@
+      testPrimitiveUnCompactingFromSource(
+          methodName: 'i64',
+          expectedValue: -2.toBigInt.pow(64 - 1),
+          compactedHex: '0x0000000000000080');
+
+      // highest acceptable possible
+      testPrimitiveUnCompactingFromSource(
+          methodName: 'i64',
+          expectedValue: 2.toBigInt.pow(64 - 1) - 1.toBigInt,
+          compactedHex: '0xffffffffffffff7f');
+    }
+
+    // i128
+    {
+      // lowest acceptable possible
+      testPrimitiveUnCompactingFromSource(
+          methodName: 'i128',
+          expectedValue: -2.toBigInt.pow(128 - 1),
+          compactedHex: '0x00000000000000000000000000000080');
+
+      // highest acceptable possible
+      testPrimitiveUnCompactingFromSource(
+          methodName: 'i128',
+          expectedValue: 2.toBigInt.pow(128 - 1) - 1.toBigInt,
+          compactedHex: '0xffffffffffffffffffffffffffffff7f');
+    }
+
+    // i256
+    {
+      // lowest acceptable possible
+      testPrimitiveUnCompactingFromSource(
+          methodName: 'i256',
+          expectedValue: -2.toBigInt.pow(256 - 1),
+          compactedHex:
+              '0x0000000000000000000000000000000000000000000000000000000000000080');
+
+      // highest acceptable possible
+      testPrimitiveUnCompactingFromSource(
+          methodName: 'i256',
+          expectedValue: 2.toBigInt.pow(256 - 1) - 1.toBigInt,
+          compactedHex:
+              '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff7f');
+    }
+
+    // Bool
+    // true
+    testPrimitiveUnCompactingFromSource(
+        methodName: 'boolean', expectedValue: true, compactedHex: '0x01');
+    // false
+    testPrimitiveUnCompactingFromSource(
+        methodName: 'boolean', expectedValue: false, compactedHex: '0x00');
+
+    testPrimitiveUnCompactingFromSource(
+        methodName: 'str',
+        expectedValue: 'github: justkawal',
+        compactedHex: '0x446769746875623a206a7573746b6177616c');
+  });
```

I'm missing a test for `EOFException` here:

`bytes`: returns an `EOFException` when the bytes length doesn't match
`str`: Same case, and also should return an error if the bytes aren't a valid String

---

Added exception test-cases ✅.